code/game/objects/items/storage/backpack.dm

```diff
@@ -82,7 +90,12 @@
 		investigate_log("has become a singularity. Caused by [user.key]", INVESTIGATE_SINGULO)
 		to_chat(user, "<span class='danger'>The Bluespace interfaces of the two devices catastrophically malfunction!</span>")
 		qdel(W)
-		var/obj/singularity/singulo = new /obj/singularity (get_turf(src))
+
+		if(emagged)
+			singulo = new /obj/singularity/boh/emagged (get_turf(src))

```

Comment:
Why subtype? why not pass the lifespan directly into a new parameter for the boh? less code needed
so:
- singulo = new /obj/singularity/boh/(get_turf(src), 200)

then in sunglarity.dm
- singularity/Initialize(var/turf, var/span = 0)
  - lifespan = span

Reply:
well i figure that lifespan might be useful for some of the existing singularity subtypes, or new ones.